packages/core/schematics/migrations/typed-forms/util.ts

```diff
@@ -0,0 +1,79 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import ts from 'typescript';
+
+import {getImportSpecifier} from '../../utils/typescript/imports';
+import {isReferenceToImport} from '../../utils/typescript/symbol';
+
+export const controlClassNames = ['AbstractControl', 'FormArray', 'FormControl', 'FormGroup'];
+export const builderMethodNames = ['control', 'group', 'array'];
+export const anySymbolName = 'AnyForUntypedForms';
+
+export interface MigratableNode {
+  node: ts.Expression;
+  generic: string;
+}
+
+export function getControlClassImports(sourceFile: ts.SourceFile) {
+  return controlClassNames.map(cclass => getImportSpecifier(sourceFile, '@angular/forms', cclass))
+      .filter(v => v != null);
+}
+
+export function getFormBuilderImport(sourceFile: ts.SourceFile) {
+  return getImportSpecifier(sourceFile, '@angular/forms', 'FormBuilder');
+}
+
+export function getAnyImport(sourceFile: ts.SourceFile) {
+  return getImportSpecifier(sourceFile, '@angular/forms', anySymbolName);
+}
+
+export function findControlClassUsages(
+    sourceFile: ts.SourceFile, typeChecker: ts.TypeChecker,
+    importSpecifier: ts.ImportSpecifier|null): MigratableNode[] {
+  if (!importSpecifier) return [];
+  let generic = `<${anySymbolName}>`;
+  if (importSpecifier.name.getText() === 'FormArray' ||
+      importSpecifier.propertyName?.getText() === 'FormArray') {
+    generic = `<${anySymbolName}[]>`;
+  }
+  const usages: MigratableNode[] = [];
+  ts.forEachChild(sourceFile, function visitNode(node: ts.Node) {
```

Comment:
Prefer lambda syntax:
```suggestion
  ts.forEachChild(sourceFile, (node: ts.Node) => {
```

Reply:
Done.